src/main/java/hu/bme/mit/spaceship/TorpedoStore.java

```diff
@@ -30,18 +30,20 @@ public TorpedoStore(int numberOfTorpedos){
 
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
     }
 
     boolean success = false;
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
-    double r = generator.nextDouble();
-
+    private Random generator = new Random();
```

Comment:
```suggestion
    private Random generator = new Random();
```ezt  nem a metodusban, hanem a classban kell letrehozni, igy nem kell mindig uj Random osztalyt letrehozni a fuggveny meghivasakor, csak ujrahasznaljuk